### DISCUSSION TITLE
Why a weird Email linked to a wrong user?

### DISCUSSION THREAD

**Discussion by X-21** [community/community#25474] [Actions]
<p>A Python script use ubuntu@localhost.localdomain as username and email for git config, and then push a commit to the remote repo. But a weird thing was happened, this commit was linked to another user(his name in github is “mountainswen”).</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/x-21">@X-21</a>,</p>
<p>I can reproduce the same behavior.</p>
<p><img src="https://user-images.githubusercontent.com/46307996/181091298-cc01f366-166c-4c50-bc8d-50246866bf59.png" alt="issue" data-base62-sha1="3O5ftf817AiqM3Ud6OIeeBF9xb2" width="645" height="279"></p>
<p>Looks like, the email address “<strong>ubuntu@localhost.localdomain</strong>” coincidentally is consistent with the account (may be a bot account) of the user “<strong>mountainswen</strong>”.</p>
<p>To avoid this situation, I recommend you change the email address to another.</p>

**Comment by X-21** [comment]
<p>Thanks. I rewrote the Python script and it’s normal now. But I am confused here, why the account can use a email address which is impossible to verify.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/x-21">@X-21</a>,</p>
<p>As far as I know, when setting the local git config, the values of “<strong>user.name</strong>” and “<strong>user.email</strong>” can be filled with any ones. Even if the user name and email are not existing.<br>
When execute the “<strong>git push</strong>” command, the commits can be pushed normally as long as the authorization token is valid and has the required access.</p>

**Comment by X-21** [comment]
<p>Yes it is. But my point is, the user “mountainswen”, he uses “ubuntu@localhost.localdomain” as his email when register a github account, how he can verify this email address? I think “localhost.localdomain” just is a linux HOSTNAME.</p>
<p>I mean, if I use “foo@localhost.localdomain” as a email address to register in github, who will receive the verify message?</p>
<p>Thank you again.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/x-21">@X-21</a>,</p>
<p>This is a question about GitHub account, I recommend that you can go to the <a href="https://github.community/c/github-account-questions/42"><strong>GitHub Account Questions board</strong></a> of the GitHub Community to look for an existing topic or start a new one about this account question.</p>

**Comment by X-21** [comment]
<p>Ok, thank you brightran.</p>

**Comment by X-21** [comment, marked as answer]
<p><a href="https://github.community/t/why-a-user-can-use-localdomain-eamil-address-as-register-email/136732/4">https://github.community/t/why-a-user-can-use-localdomain-eamil-address-as-register-email/136732/4?u=x-21</a></p>
<blockquote>
<p>If you add an email to your account, and commit using that email, it will be attributed to your account. It doesn’t need to be verified.</p>
</blockquote>

**Comment by nathub** [comment]
Can this be fixed in the history if I've already commited and pushed to github, using the wrong email?

i just discovered one of my previous commits did this last month. now my repo says I have two contributors, and I don't want the mystery person listed anymore.
